### PR TITLE
Openshift apiserver 4.13 kubernetes 1.26 rebase

### PR DESCRIPTION
**Pull request by dinhxuanvu**
Carry upstream batches from 1.25 branch to 1.26

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
@soltysh PTAL when you have the time.
I'm doing a 1.26 rebase for openshift-apiserver and according to the Luis, this PR is the first step. Thanks.

**Comment by dinhxuanvu** [comment]
Just an FYI, there were a few merge conflicts with some commits and I did manually resolve them. Nothing was major as far as I can tell. The two commits that stand out on the conflict were:
- UPSTREAM: <carry>: simplify the authorizer patch to allow the flags to function
- UPSTREAM: <carry>: create termination events

File: pkg/server/config.go [hunk 1]
```diff
@@ -1077,6 +1079,11 @@ func AuthorizeClientBearerToken(loopback *restclient.Config, authn *Authenticati
 
 	tokenAuthenticator := authenticatorfactory.NewFromTokens(tokens, authn.APIAudiences)
 	authn.Authenticator = authenticatorunion.New(tokenAuthenticator, authn.Authenticator)
+
+	if !skipSystemMastersAuthorizer {
+		tokenAuthorizer := authorizerfactory.NewPrivilegedGroups(user.SystemPrivilegedGroup)
+		authz.Authorizer = authorizerunion.New(tokenAuthorizer, authz.Authorizer)
+	}
```
[the remark this answers is not in the document]
Comment (dinhxuanvu) [reply pkg/server/config.go:1086]: Sounds good. Will do.

**Comment by dinhxuanvu** [comment]
@sanchezl PTAL. Please double-check this commit: [UPSTREAM: <carry>: openshift-apiserver: add RemoveOpenAPIData method …](https://github.com/openshift/kubernetes-apiserver/pull/38/commits/ea44d3ac465abdac4b74a9fc4e9a2d00f6d9e662)

File: pkg/server/config.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanchezl) [reply pkg/server/config.go]: l do not think we need this patch at all, drop it.

**Review by sanchezl — APPROVED** [review]

**Pull request merged**